src/sail_json_backend/json.ml

```diff
@@ -398,14 +398,77 @@ let remove_identity_funcs op =
       else op
   ) op identity_funcs
 
+let optional_operand k op =
+  let pattern = Str.regexp ("maybe_[A-Za-z_]+\\(" ^ "(" ^ op ^ ")" ^ "\\)") in
```

Comment:
I presume Sail functions can have digits as well as upper/lower and underscore? If so, you'll need to add those characters to your regex.

Also, let's add a comment here, since this is slightly hacky in depending on a very particular function name pattern.

Reply:
Oops. Fixed now. Also added the explanatory comment.